packages/code-editor/src/CodeDisplay/types.ts

```diff
@@ -0,0 +1,48 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2021 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+export interface CodeDisplayProps {
+  code: string
+  language?: string
+  pattern?: string
+  transparent?: boolean
+  inline?: boolean
+  lineNumbers?: boolean
+}
+
+export type TokenOutputProps = {
+  key?: React.Key
+  style?: any
+  className: string
+  children: string
+  [otherProp: string]: any
+}
+
+export interface LineItemProps {
+  index: number
+  tokenProps: TokenOutputProps
+  pattern: string
+}
```

Comment:
Copies of unexported Prism types

Reply:
Pity they don't export them. Maybe we should create a PR for Prism?

Reply:
any particular reason types are in their own file for the CodeDisplay component? In `api-explorer` and `run-it` types and component implementations coexist in the same file